web/impact/impact/v1/helpers/matching_industry_criterion_helper.py

```diff
@@ -18,14 +18,13 @@ def __init__(self, subject):
         super().__init__(subject)
         self._top_level_id_cache = None
 
-    @classmethod
-    def _industry_map(cls):
-        if cls.industries is None:
-            cls.industries = cls.instances_by_name(Industry)
-        return cls.industries
+    def _industry_map(self):
+        if self.industries is None:
```

Comment:
what is the use of this method if its now an instance method? the _industry_map method seems to only be referenced once so i dont think it needs to be cached.

Reply:
This instance is persistent, so I suspect it's hit multiple times in the course of a given analysis. (once for each of the top-level industries)

Reply:
we can set self.industry_map on the __init__ method and getting rid of this method

Reply:
Per our standards, we avoid db activity in the __init__

Reply:
What I wonder here is how this code change relates to the solution itself, I understand everything else but this as far as the ticket.

Reply:
@VincentHokie 
> Discovered another caching-on-the-class issue while writing a test for this. Closing while I look into that.

As written this code made it impossible to write more than one test for any given matching criterion.

Reply:
ah! because it was stored as a class variable and not on the object the cache was available cross-test..I see